File: src/main/java/net/igneo/icv/entity/weapon/snakeBite/SnakeBiteModel.java
package net.igneo.icv.entity.weapon.snakeBite;

import net.igneo.icv.ICV;
import net.minecraft.resources.ResourceLocation;
import software.bernie.geckolib.model.GeoModel;

public class SnakeBiteModel extends GeoModel<SnakeBiteEntity> {
    private final ResourceLocation model = new ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json");
    private final ResourceLocation texture = new ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png");
    private final ResourceLocation animations = new ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json");


    @Override
    public ResourceLocation getModelResource(SnakeBiteEntity snakeBiteEntity) {
        return this.model;
    }

    @Override
    public ResourceLocation getTextureResource(SnakeBiteEntity snakeBiteEntity) {
        return this.texture;
    }

    @Override
    public ResourceLocation getAnimationResource(SnakeBiteEntity snakeBiteEntity) {
        return this.animations;
    }
}

File: src/main/java/net/igneo/icv/entity/weapon/snakeBite/SnakeBiteModel.kt
package net.igneo.icv.entity.weapon.snakeBite

import net.igneo.icv.ICV
import net.minecraft.resources.ResourceLocation
import software.bernie.geckolib.model.GeoModel

class SnakeBiteModel : GeoModel<SnakeBiteEntity?>() {
    private val model = ResourceLocation(ICV.MOD_ID, "geo/placeholder.geo.json")
    private val texture = ResourceLocation(ICV.MOD_ID, "textures/entity/placeholder.png")
    private val animations = ResourceLocation(ICV.MOD_ID, "animations/placeholder.animation.json")


    override fun getModelResource(snakeBiteEntity: SnakeBiteEntity?): ResourceLocation {
        return this.model
    }

    override fun getTextureResource(snakeBiteEntity: SnakeBiteEntity?): ResourceLocation {
        return this.texture
    }

    override fun getAnimationResource(snakeBiteEntity: SnakeBiteEntity?): ResourceLocation {
        return this.animations
    }
}
